doc/guides/dts/howtos.rst

```diff
@@ -307,104 +307,262 @@ Assuming you have a suitable device driver associated with the
 enable the driver via Kconfig and :ref:`get the struct device <dt-get-device>`
 for your newly added bus node, then use it with that driver API.
 
-.. _dt-driver-howto:
+.. _dt-create-devices:
 
-Create struct devices in a driver
-*********************************
+Write device drivers using devicetree APIs
+******************************************
 
-If you're writing a device driver, it should be devicetree aware so that
-applications can configure it and access devices as described above. In short,
-you must create a ``struct device`` for every enabled instance of the
-compatible that the device driver supports, and set each device's name to the
-``DT_LABEL()`` of its devicetree node.
+"Devicetree-aware" :ref:`device drivers <device_model_api>` should create a
+``struct device`` for each enabled devicetree node with a particular
+:ref:`compatible <dt-important-props>` (or related set of compatibles)
+supported by the driver. Here, "enabled" means the node's ``status`` property
+is set to ``"okay"``.
```

Comment:
This is not the case right now (it's status != disabled instead), but I think making an explicit "okay" required to enable the device is probably what we have time to do by 2.3. Unless we want to try to pull in https://github.com/zephyrproject-rtos/zephyr/issues/19448 for the release. I've lost the bubble on whether that's still on the table, but the issue isn't tagged for the release.